packages/inferno-router/src/doAllAsyncBefore.ts

```diff
@@ -2,16 +2,13 @@ export default function doAllAsyncBefore(renderProps) {
   const promises: Object[] = [];
 
   const getAsyncBefore = function(root) {
-    if (root) {
-      if (root.props && root.props.children) {
-        getAsyncBefore(root.props.children);
+    if (root && root.props) {
+      if (root.props.children) {
+          getAsyncBefore(root.props.children);
       }
-
-      if (root.type.name === "Route" && root.props.asyncBefore) {
-        // Resolve asyncBefore
-        promises.push(
-          root.type.prototype.doAsyncBefore.call(root, root.props.params)
-        );
+      if (typeof root.props.asyncBefore === 'function' && root.type.prototype.doAsyncBefore) {
```

Comment:
It was previously reading that property from props now its reading it from prototype, isn't this breaking change? Also what about v4 router in dev?

Reply:
It is reading from both now. The issue was that it was doing `root.type.name === "Route"` but this property is renamed in a production build. So I changed the check to `(typeof root.props.asyncBefore === 'function' && root.type.prototype.doAsyncBefore)` which should survive the minification process but gives the same net effect.

The v4 router doesn't have the same async capabilities yet as far as I know. I need to dig into the code and see what I can do about it.